eng/pipelines/ref-updater.yml

```diff
@@ -0,0 +1,50 @@
+# Updates repository Refs on all pipelines
+resources:
+  repositories:
+    - repository: azure-sdk-for-net
+      type: github
+      name: Azure/azure-sdk-for-net
+      endpoint: azure
+    - repository: azure-sdk-for-js
+      type: github
+      name: Azure/azure-sdk-for-js
+      endpoint: azure
+    - repository: azure-sdk-for-java
+      type: github
+      name: Azure/azure-sdk-for-java
+      endpoint: azure
+    - repository: azure-sdk-for-python
+      type: github
+      name: Azure/azure-sdk-for-python
+      endpoint: azure
+    - repository: azure-sdk-build-tools
+      type: git
+      name: internal/azure-sdk-build-tools
+
+trigger: none
+
+pr: none
+
+pool:
+  vmImage: windows-2019
+
+jobs:
+  - job: UpdateRepoResourceRef
+    displayName: Update Repository Resource Ref
+
+    steps:
+    - checkout: self
```

Comment:
I'm not sure about this multiple checkout approach as we need to do more steps then just checkout we will also need to submit pull requests for each repo. Take a look at how I did that in https://github.com/Azure/azure-sdk-tools/blob/master/eng/pipelines/eng-common-sync.yml I think we can use a similar pattern here.